### PR TITLE
Add information about staging repos to README

### PR DESCRIPTION
**Pull request by pmorie**
Pretty much what it sounds like.

### CODE REVIEW

File: staging/README.md [hunk 1]
```diff
@@ -1,16 +1,45 @@
...
+
+- [`k8s.io/apimachinery`](https://github.com/kubernetes/apimachinery)
+- [`k8s.io/apiserver`](https://github.com/kubernetes/apiserver)
+- [`k8s.io/client-go`](https://github.com/kubernetes/client-go)
+- [`k8s.io/kube-aggregator`](https://github.com/kubernetes/kube-aggregator)
+- [`k8s.io/sample-apiserver`](https://github.com/kubernetes/sample-apiserver)
+
+## Sources of truth
+
+Every repository except `client-go` in this directory is authoritative, i.e.
```
Comment (deads2k) [inline staging/README.md]: @caesarxuchao this will need to be updated as soon as 1.8 starts, right?

Comment (caesarxuchao) [inline staging/README.md]: In the early phase of 1.8. https://github.com/kubernetes/kubernetes/pull/44784 will get rid of the copy of the types. clientset/informers/listers will still be copied until i send another PR to let the code-gen directly generate code to client-go. That should be a relatively easy PR.

File: staging/README.md [hunk 2]
```diff
@@ -1,16 +1,45 @@
...
-`k8s.io/<package-name>`, as if the package were vendored. Packages will be
-vendored from `k8s.io/<package-name>` for real after the test matrix is
-converted to vendor k8s components.
+Repositories currently staged here:
+
+- [`k8s.io/apimachinery`](https://github.com/kubernetes/apimachinery)
+- [`k8s.io/apiserver`](https://github.com/kubernetes/apiserver)
+- [`k8s.io/client-go`](https://github.com/kubernetes/client-go)
+- [`k8s.io/kube-aggregator`](https://github.com/kubernetes/kube-aggregator)
+- [`k8s.io/sample-apiserver`](https://github.com/kubernetes/sample-apiserver)
```
Comment (caesarxuchao) [inline staging/README.md]: Missed the apiextensions repo.

File: staging/copy.sh [hunk 3]
```diff
@@ -147,7 +148,8 @@ cp -r "${CLIENT_REPO_TEMP}"/vendor/k8s.io/kubernetes/* "${CLIENT_REPO_TEMP}"/
 rm -rf "${CLIENT_REPO_TEMP}"/vendor
 
 echo "rewriting Godeps.json"
-# The entries for k8s.io/apimahcinery are not removed from Godeps.json, though
+
```
Comment (caesarxuchao) [inline staging/copy.sh]: Could you keep this file unchanged? https://github.com/kubernetes/kubernetes/pull/44784 will get rid of most of this script.

Comment (pmorie) [inline staging/copy.sh]: yeah, i totally can do that

File: staging/README.md [hunk 4]
```diff
@@ -1,16 +1,46 @@
...
+
+## Sources of truth
+
+Every repository except `client-go` in this directory is authoritative, i.e.
+the only copy of the code.  When changes are needed to these repositories, the
+changes should be made here.
+
+The packages in `staging/src/k8s.io/client-go/pkg` are copied from `pkg/` in
+the kubernetes project. If you modify the original code in `pkg/`, you need to
+run `hack/godep-restore.sh` from the k8s root directory, followed by `hack
```
Comment (caesarxuchao) [inline staging/README.md]: Running `hack/godep-restore.sh` before `hack/update-staging-client-go` is not required now.

File: staging/README.md [hunk 5]
```diff
@@ -1,16 +1,46 @@
...
+- [`k8s.io/kube-aggregator`](https://github.com/kubernetes/kube-aggregator)
+- [`k8s.io/sample-apiserver`](https://github.com/kubernetes/sample-apiserver)
+
+## Sources of truth
+
+Every repository except `client-go` in this directory is authoritative, i.e.
+the only copy of the code.  When changes are needed to these repositories, the
+changes should be made here.
+
+The packages in `staging/src/k8s.io/client-go/pkg` are copied from `pkg/` in
```
Comment (caesarxuchao) [inline staging/README.md]: `staging/src/k8s.io/client-go/<informer/listers/kubernetes>` are also copied currently. They will be authoritative in one or two weeks.

File: staging/README.md (hunk 1 above)
Comment (caesarxuchao) [inline staging/README.md]: client-go is authoritative now. I recently updated the README for that fact.

File: staging/README.md (hunk 4 above)
Comment (caesarxuchao) [inline staging/README.md]: Please ignore my comment. hack/update-staging-client-go.sh is removed.

File: staging/README.md (hunk 5 above)
Comment (caesarxuchao) [inline staging/README.md]: Please ignore. They are authoritative now.

**Comment by pmorie** [comment]
@caesarxuchao @deads2k sorry for the latency - updated.

File: staging/README.md [hunk 6]
```diff
@@ -1,12 +1,40 @@
...
-the code. You can directly modify such code.
+Repositories currently staged here:
+
+- [`k8s.io/apiextensions-apiserver`](https://github.com/kubernetes/apiextensions-apiserver)
+- [`k8s.io/api`](https://github.com/kubernetes/api)
+- [`k8s.io/apimachinery`](https://github.com/kubernetes/apimachinery)
+- [`k8s.io/apiserver`](https://github.com/kubernetes/apiserver)
+- [`k8s.io/client-go`](https://github.com/kubernetes/client-go)
+- [`k8s.io/kube-aggregator`](https://github.com/kubernetes/kube-aggregator)
+- [`k8s.io/kube-gen`](https://github.com/kubernetes/kube-gen)
```
Comment (caesarxuchao) [inline staging/README.md]: This one is not published yet, so how about [`k8s.io/kube-gen`](https://github.com/kubernetes/kube-gen)(to be published)

File: staging/README.md [hunk 7]
```diff
@@ -1,12 +1,40 @@
...
+Kubernetes code imports a package from the `k8s.io/client-go` repository, that
+import is resolved to `staging/src/k8s.io/client-go` relative to the project
+root:
+
+```go
+// pkg/example/some_code.go
+package example
+
+import (
+  "k8s.io/client-go/pkg/api/v1" // resolves to staging/src/k8s.io/client-go/pkg/api/v1
```
Comment (caesarxuchao) [inline staging/README.md]: This package doesn't exist anymore, maybe use "k8s.io/client-go/dynamic" as an example.

**Review by caesarxuchao — COMMENTED** [review]
Two nits and lgtm.

File: staging/README.md (hunk 7 above)
Comment (pmorie) [inline staging/README.md]: whoops!

**Comment by caesarxuchao** [comment]
Thanks @pmorie!
